packages/core/src/logger/logger.ts

```diff
@@ -1,144 +1,196 @@
-import { AsyncLocalStorage } from 'async_hooks';
-import { QueueManager } from '../queues';
-import { LogLevel } from '../types';
-import { PluginId } from '../plugins';
-import { LoggerInterface } from './loggerInterface';
-
-export class Logger implements LoggerInterface {
-  private readonly queueManager: QueueManager;
-
-  private readonly globalLogger?: LoggerInterface;
+/* eslint-disable guard-for-in */
+import { v4 } from 'node-uuid';
 
-  private readonly context: AsyncLocalStorage<string>;
-
-  constructor(queueManager: QueueManager, globalLogger?: LoggerInterface) {
-    this.queueManager = queueManager;
-    this.globalLogger = globalLogger;
-    this.context = new AsyncLocalStorage();
+import { AsyncLocalStorage } from 'async_hooks';
+import { LogLevel, TContextBase, LogEntry } from './types';
+import { LogHandler } from './handler';
+
+export class Logger<
+  TContext extends TContextBase = { processId: string },
+  TGlobalContext extends Record<string, any> = {},
+> {
+  private readonly globalContext: TGlobalContext;
+
+  private readonly logHandlers: LogHandler<Partial<TContext>, TGlobalContext>[];
+
+  private readonly asyncStorage: AsyncLocalStorage<Partial<TContext>>;
```

Comment:
naming: nejdřív jsem začal psát něco o tom, že Logger možná nepotřebuje používat ALS, protože už si nepotřebuje nic trackovat v rámci requestu / tasku; pak jsem si teprve všimnul, že tahle ALS drží `TContext`.. možná by stálo za to sjednotit & zlepšit naming věcí, aby to bylo trochu víc obvious - napadá mě něco takovýhleho:
 - `TContext` -> `TTaskContext` (aby se odlišil od `TGlobalContext`)
 - `fork()` a `ensureFork()` -> `runTask()` (chování `ensureFork()` může bejt zapouzdřený ve stejný funkci pomocí explicitního `force: false`)

bude podle mě jednodušší v dokumentaci vysvětlit koncept "tasku" a nemíchat do toho nějaký "forky", což jsem převzal z MikroORM, ale přijde mi, že to tady není tak sémanticky krásný

Reply:
yes, done